### PR TITLE
Fix typo in param and correct german to_currency params.

### LINKED ISSUE #148: Wrong function call for currency in German
### Expected Behaviour

from num2words import num2words

num2words(20.18, to='currency', lang='de')

==> zwanzig euro und 18 cent.

### Actual Behaviour

TypeError: to_currency() got an unexpected keyword argument 'jointxt'

### Explanation

When trying convert currencies in German, super(Num2Word_EU, self).to_currency(val, jointxt="und", longval=longval) is called. However, it should be self.to_splitnum(val, hightxt="euro", lowtxt="cent", jointxt="und", longval=longval).

### PR DESCRIPTION
**Pull request by erozqba**
### Changes proposed in this pull request:

* Fix typo in code
* Fix bug in german to_currency params.

### Status

- [X] READY
- [ ] HOLD
- [ ] WIP (Work-In-Progress)

### How to verify this change

Run the tests.

### Additional notes

*If applicable, explain the rationale behind your change.*

### CODE REVIEW

**Comment by shulcsm** [comment]
Sorry for spelling separator wrong :<

**Comment by erozqba** [comment]
@shulcsm no worries! I was the one that has do the review of the changes and has accepted it without notice it. Do you know German? We need more tests here!

**Comment by sbadecker** [comment]
@erozqba I do now German :) What exactly do you need, just a couple of more test values + their text representation as test cases?

**Comment by erozqba** [comment]
@sbadecker Yes, that all we need, if you look into the other languages tests you could find a lot of examples of different tests, so we only need to do something similar for German.

**Comment by erozqba** [comment]
@sbadecker Do you think you will be able to find some time to help me here with the tests? Thanks in advance.

**Comment by sbadecker** [comment]
@erozqba Sure, anything specific you would like being tested?

**Comment by erozqba** [comment]
@sbadecker you could check the existing test in the file tests/test_fr_br.py to get some inspiration for possible tests cases?

File: tests/test_de.py [hunk 1]
```diff
@@ -65,3 +65,23 @@ def test_ordinal_for_negative_numbers(self):
 
     def test_ordinal_for_floating_numbers(self):
         self.assertRaises(TypeError, num2words, 2.453, ordinal=True, lang='de')
+
+    def test_to_currency_for_some_numbers(self):
+        self.assertEqual(
+            num2words(190, lang='de', to='currency'),
+            "eins euro und neunzig cents"
```
Comment (hlef) [inline tests/test_de.py]: Looks wrong to me. 190 is "einhundertneunzig Euro null Cent" or "hundertneunzig Euro null Cent". Also, the "null Cent" sounds weird. FYI "ein Euro neunzig Cent" means 1.90 euro.

File: tests/test_de.py [hunk 2]
```diff
@@ -65,3 +65,23 @@ def test_ordinal_for_negative_numbers(self):
...
+        )
+        self.assertEqual(
+            num2words(100.0, lang='de', to='currency'),
+            "hundert euro und null cents"
```
Comment (hlef) [inline tests/test_de.py]: I would rather say "einhundert Euro null Cent"

File: tests/test_de.py [hunk 3]
```diff
@@ -65,3 +65,23 @@ def test_ordinal_for_negative_numbers(self):
...
+        )
+
+    def test_to_currency_for_floating_numbers(self):
+        self.assertEqual(
+            num2words(3.4, lang='de', to='currency'),
+            "drei euro und vierzig cents"
```
Comment (hlef) [inline tests/test_de.py]: "drei Euro vierzig Cent" ?

File: tests/test_de.py [hunk 4]
```diff
@@ -65,3 +65,23 @@ def test_ordinal_for_negative_numbers(self):
...
+        )
+        self.assertEqual(
+            num2words(20.18, lang='de', to='currency'),
+            "zwanzig euro und achtzehn cents"
```
Comment (hlef) [inline tests/test_de.py]: "zwanzig Euro achtzehn Cent" ?

**Comment by erozqba** [comment]
@hlef Could you provide a patch to correct these issues with German?

**Comment by sebastianrakel** [comment]
I'm a native german and can review also after the PR is splitted. Need this fix :D

**Comment by sebastianrakel** [comment]
@hlef is there any progress in seperate the issues from this PR?

**Comment by hlef** [comment]
Working on it on my fork, see https://github.com/hlef/num2words.

I'll PR once ready.

**Comment by hlef** [comment]
See #222. Review from german natives highly appreciated !

**Pull request closed without merging**